src/domain/request/usecase/ExecuteReadyRequestsUseCase.js

```diff
@@ -0,0 +1,45 @@
+class ExecuteReadyRequestsUseCase {
+  constructor(
+    fetchDataUseCase,
+    selectDataUseCase,
+    sendResponseToOracleUseCase,
+    requestRepository,
+    responseRepository,
+    logger,
+  ) {
+    this.fetchDataUseCase = fetchDataUseCase;
+    this.selectDataUseCase = selectDataUseCase;
+    this.sendResponseToOracleUseCase = sendResponseToOracleUseCase;
+    this.requestRepository = requestRepository;
+    this.responseRepository = responseRepository;
+    this.logger = logger;
+  }
+
+  async executeReadyRequests() {
+    const requests = await this.requestRepository.getReadyRequests();
+
+    const promises = requests.map(async (request) => {
+      request.state.markAsProcessed();
+      this.logger.info(`Request ${request.id} marked as processed`);
+      try {
+        let response = await this.fetchDataUseCase.fetchDataForRequest(request);
```

Comment:
I don't like this reassignment very much. Let's see how this code changes when the sub-use-cases don't operate on the request/response level where possible.